cypress/integration/crd.spec.js

```diff
@@ -0,0 +1,167 @@
+describe('Custom Resource Definitions', () => {
+  before(() => {
+    cy.visit('/');
+  });
+
+  it('crd list is initially empty', () => {
+    cy.get('[data-cy=sidebar-crd]').click();
+    cy.url().should('include', '/#/customresourcedefinition');
+
+    cy.get('[data-cy=zero-state]').should('be.visible');
+  });
+
+  it('create a crd', () => {
+    cy.fixture('crd.yaml').then(crd => {
+      cy.createResource({content: crd});
+      cy.reload();
+    });
+  });
+
+  it('crd list should contain the new crd', () => {
+    cy.get('[data-cy=zero-state]').should('not.be.visible');
+    cy.get('kd-crd-list').within(() => {
+      cy.get('mat-row').should('have.length', 1);
+    });
+  });
+
+  it('go to crd details page', () => {
+    cy.get('kd-crd-list').within(() => {
+      cy.get('a')
+        .first()
+        .click();
+    });
+    cy.url().should(
+      'include',
+      '/#/customresourcedefinition/foos.samplecontroller.k8s.io?namespace=default',
+    );
+
+    // loads correctly
+    cy.get('kd-object-meta').within(() => {
+      cy.get('kd-property')
+        .first()
+        .contains('foos.samplecontroller.k8s.io');
+    });
+
+    // has resource information
+    cy.get('[data-cy=resource-information]').within(() => {
+      cy.get('kd-property').should('have.length', 3);
+
+      cy.get('kd-property')
+        .eq(0)
```

Comment:
We cannot assume order here, please use text matching or add IDs to the template.

Reply:
Done